CMakeLists.txt

```diff
@@ -93,11 +92,16 @@ else()
 endif()
 
 add_library(onnx_proto ${PROTO_SRCS} ${PROTO_HDRS})
-add_dependencies(onnx_proto protobuf::libprotobuf)
-target_compile_options(onnx_proto PRIVATE /WX- ${DLLEXPORT_OPTION})
+target_link_libraries(onnx_proto ${PROTOBUF_LIBRARIES})
+if(MSVC)
+    target_compile_options(onnx_proto PRIVATE /WX- ${DLLEXPORT_OPTION})
+endif()
 
 add_library(onnx ${onnx_src})
-add_dependencies(onnx onnx_proto)
```

Comment:
I think we need this one because onnx libraries will include proto headers.

Reply:
I'm apparently not knowledgeable on cmake :-) But it works in my osx and linux environments, so I guess `target_link_dependencies` implicitly adds the dependencies' include directories.

Reply:
It works because the ${ONNX_ROOT} is included when doing compilation and it's set as link dependency with target_link_dependencies.